controller/space_template.go

```diff
@@ -0,0 +1,144 @@
+package controller
+
+import (
+	"encoding/base64"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/jsonapi"
+	"github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/rest"
+	"github.com/almighty/almighty-core/spacetemplate"
+	"github.com/goadesign/goa"
+)
+
+const APIStringType = "spacetemplates"
+
+// SpaceTemplateController implements the space_template resource.
+type SpaceTemplateController struct {
+	*goa.Controller
+	db     application.DB
+	config SpaceTemplateControllerConfiguration
+}
+
+// SpaceTemplateControllerConfiguration the configuration for the SpaceTemplateController
+type SpaceTemplateControllerConfiguration interface {
+	GetCacheControlSpaceTemplates() string
+}
+
+// NewSpaceTemplateController creates a space_template controller.
+func NewSpaceTemplateController(service *goa.Service, db application.DB, config SpaceTemplateControllerConfiguration) *SpaceTemplateController {
+	return &SpaceTemplateController{
+		Controller: service.NewController("SpaceTemplateController"),
+		db:         db,
+		config:     config,
+	}
+}
+
+// Create runs the create action.
+func (c *SpaceTemplateController) Create(ctx *app.CreateSpaceTemplateContext) error {
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+
+	newSpaceTemplate := spacetemplate.SpaceTemplate{
+		Name:        *ctx.Payload.Data.Attributes.Name,
+		Description: ctx.Payload.Data.Attributes.Description,
+	}
+
+	// If the user passed in an ID, we set it
+	if ctx.Payload.Data.ID != nil {
+		newSpaceTemplate.ID = *ctx.Payload.Data.ID
+	}
+
+	// Decode the template from base64
+	templateBytes, err := base64.StdEncoding.DecodeString(*ctx.Payload.Data.Attributes.Template)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("template", ctx.Payload.Data.Attributes.Template).Expected("base64-decodable string"))
+	}
+	newSpaceTemplate.Template = string(templateBytes)
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		err = appl.SpaceTemplates().Create(ctx, &newSpaceTemplate)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
We can see how it works, but I can see that we miss accurate error messages and the properties used in the function's call that failed :/